City-of-Helsinki/helsinki-paatokset
# UHF-9055: Decision seo

# [UHF-9055](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9055)


## What was done


* Add decision nodes to sitemap.xml
* Improve metatags on decision and case pages.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-9055-decision-seo`
  * `make new shell`
* Import some decisions and cases to test with.
* Run `drush ssg` (generate sitemap.xml)

## How to test


* [ ] Check metatags on translated case `entity.node.canonical`
      https://helsinki-paatokset.docker.so/fi/asia/hel-2023-002431?paatos=4d1330df-5e7f-451d-90ab-2e739b558126
* [ ] Check metatags on untranslated case (`paatokset_decision.en`)
      https://helsinki-paatokset.docker.so/sv/arende/hel-2023-002431?beslut=26e36cc0-cacd-44c5-a254-f711e3f55c12
      - Note: all cases have `fi` langcode.
      - Canonical tag should point to the language of the decision node, `sv` in this case, although the case itself is in `fi`.
* [ ] Check metatags on translated decision `entity.node.canonical`
      - https://helsinki-paatokset.docker.so/fi/asia/hel-2023-002431/4d1330df-5e7f-451d-90ab-2e739b558126
* [ ] Check metatags on untranslated decision `paatokset_case.en`
      https://helsinki-paatokset.docker.so/en/case/hel-2023-002431?decision=4d1330df-5e7f-451d-90ab-2e739b558126
      - Canonical tag should point to the language of the decision node.
* [ ] Check that sitemap.xml file contains decision.
      https://helsinki-paatokset.docker.so/sitemap.xml

You can use this snippet to check specific metatags quicly, just paste it to JS console: `document.querySelectorAll('[rel="canonical"]').forEach(console.log) `

## Designers review



## Other PRs


* Link to other PR
